### ISSUE TITLE
DBIx::Class test failure

### ISSUE THREAD

**Issue by pikajude** [NixOS/nixpkgs#6837]
https://gist.github.com/joelteon/d2ad35f0734349593bb9

This package is a dependency of Hydra.

**Comment by wizeman** [comment]
I just want to mention that I ran into exactly the same issue.

**Comment by wizeman** [comment]
This should be fixed when DBIxClass is updated to version 0.082820, as it contains the following fix:
https://github.com/dbsrgits/dbix-class/commit/cff17b977312a6ebe079049603df300397c030d9

I will submit a PR soon.

**Comment by wizeman** [comment]
It turns out version `0.082820`'s tests fail in `t/52leaks.t`... *sigh*

**Comment by abbradar** [comment]
I have also run into this. I've tried to update several dependencies of DBIxClass too, it haven't helped. Master branch of the package fails, too -- it should be some dependency I think but updating them manually is laborious process. Too bad we don't have perl-ng! ^_^

**Comment by abbradar** [comment]
@wizeman, I see you've disabled check phase in this package locally -- do you have any problems? I'm looking for a workaround ^_^

**Comment by wizeman** [comment]
@abbradar I think it should be fine to disable the check phase if you're using the same DBIxClass version that is currently committed to nixpkgs (not the updated version).

As far as I can remember, the `t/100populate.t` test only fails because today's date is >= March 2015.
If I understood the issue correctly, they wanted to disable some warnings after this date, so they made a test that fails loudly if they hadn't disabled those warnings by this date. Or something similar, I'm not sure to be honest.

Basically, the fix was to remove the warning and the test. Since this was the only failure in the test suite, it should be harmless.

**Comment by abbradar** [comment]
Good, thanks -- maybe we want to add the patch to master for now until the proper fix is here? Sounds harmless and would fix Hydra builds, too.

**Comment by abbradar** [comment]
On the second thought it would break developers' promise [1], if we want to add workaround to master, maybe we should disable check phase instead.
[1]: https://github.com/dbsrgits/dbix-class/commit/cff17b977312a6ebe079049603df300397c030d9#diff-58187a4d7d12ff7fc918ce87cf96a42dL64

**Comment by seppeljordan** [comment]
Does anybody have a workarround for this issue that can be applied to nixos/configure.nix?

**Comment by konobi** [comment]
I beliebe upgrading the dependency of DBD-SQLite to greater than 1.38_01 should fix this.

**Comment by konobi** [comment]
Oh... "The warning was removed in version 0.082900"

**Comment by konobi** [comment]
https://metacpan.org/changes/distribution/DBIx-Class#L21 shows it as removed in a newer version, should give this version a try?

**Comment by wizeman** [comment]
@konobi That newer version has failure in another test (I haven't investigated why, though).

**Comment by konobi** [comment]
There'll be a new version tomorrow that will fix the find leaks issue.

**Comment by pSub** [comment]
Fixed in 5bc1ddd78ffe566dc3f36b6fecb14fd0b0a3ac8f.
